python/lsst/sims/maf/metrics/periodicDetectMetric.py

```diff
@@ -1,72 +1,96 @@
+        super(PeriodicDetectMetric, self).__init__([mjdCol, m5Col, filterCol], metricName=metricName,
+                                                   units='N Detected (0, %i)' % np.size(periods), **kwargs)
 
     def run(self, dataSlice, slicePoint=None):
         result = 0
-        lc = self.amplitude*np.sin(dataSlice[self.mjdCol]*(np.pi*2)/self.period) + self.starMag
-        n_pts = lc.size
-        # If we had a correct model with phase, amplitude, period, zeropoint, then chi_squared/DoF would be ~1 with 4 free parameters.
-        # The mean is one free parameter
+        n_pts = np.size(dataSlice[self.mjdCol])
+        n_filt = np.size(np.unique(dataSlice[self.filterCol]))
 
-        p1 = 1.
-        p2 = 4.
+        # If we had a correct model with phase, amplitude, period, mean_mags, then chi_squared/DoF would be ~1 with 3+n_filt free parameters.
+        # The mean is one free parameter
+        p1 = n_filt
+        p2 = 3.+n_filt
         chi_sq_2 = 1.*(n_pts-p2)
 
+        u_filters = np.unique(dataSlice[self.filterCol])
+
         if n_pts > p2:
-            snr = m52snr(lc, dataSlice[self.m5Col])
-            delta_m = 2.5*np.log10(1.+1./snr)
-            weights = 1./(delta_m**2)
-            weighted_mean = np.sum(weights*lc)/np.sum(weights)
-            chi_sq_1 = np.sum(((lc - weighted_mean)**2/delta_m**2))
-            # Yes, I'm fitting magnitudes rather than flux. At least I feel kinda bad about it.
-            # F-test for nested models:  https://en.wikipedia.org/wiki/F-test
-            # https://stackoverflow.com/questions/21494141/how-do-i-do-a-f-test-in-python/21503346
-            f_numerator = (chi_sq_1 - chi_sq_2)/(p2-p1)
-            f_denom = chi_sq_2/(n_pts-p2)
-            f_val = f_numerator/f_denom
+            for period, starMag, amplitude in zip(self.periods, self.starMags, self.amplitudes):
+                chi_sq_1 = 0
+                mags = utils.stellarMags(self.SedTemplate, rmag=starMag)
+                for filtername in u_filters:
+                    in_filt = np.where(dataSlice[self.filterCol] == filtername)[0]
+                    lc = amplitude*np.sin(dataSlice[self.mjdCol][in_filt]*(np.pi*2)/period) + mags[filtername]
+                    snr = m52snr(lc, dataSlice[self.m5Col][in_filt])
+                    delta_m = 2.5*np.log10(1.+1./snr)
+                    weights = 1./(delta_m**2)
+                    weighted_mean = np.sum(weights*lc)/np.sum(weights)
+                    chi_sq_1 += np.sum(((lc - weighted_mean)**2/delta_m**2))
+                # Yes, I'm fitting magnitudes rather than flux. At least I feel kinda bad about it.
+                # F-test for nested models:  https://en.wikipedia.org/wiki/F-test
+                # https://stackoverflow.com/questions/21494141/how-do-i-do-a-f-test-in-python/21503346
+                f_numerator = (chi_sq_1 - chi_sq_2)/(p2-p1)
+                f_denom = chi_sq_2/(n_pts-p2)
```

Comment:
Am I missing something here though? 
chi_sq_2 = 1 * (n_pts-p2)  (line 67)
and then f_denom = chi_sq_2 / (n_pts-p2)  .. 
Is this a typo or is f_denom always supposed to be 1?